courses/free-courses-si.md

```diff
@@ -1,30 +1,56 @@
 ### Index
 
+* [Algorithms & Data Structures](#algorithms--data-structures)
 * [ASP.NET Core](#aspnet-core)
 * [C#](#csharp)
+* [Deep Learning](#deep-learning)
+* [Docker](#docker)
+* [Flutter](#flutter)
 * [HTML and CSS](#html-and-css)
 * [Java](#java)
     * [Spring Boot](#spring-boot)
 * [JavaScript](#javascript)
     * [React](#react)
+* [Machine Learning](#machine-learning)
 * [PHP](#php)
 
 
+### Algorithms & Data Structures
+
+* [Data Structures and Algorithms | Sinhala](https://www.youtube.com/playlist?list=PL495mke12zYDIwsabzb61OLdBpg3QDcXg) - CodePRO LK
+
+
 ### ASP.NET Core
 
 * [WEB API-ASP.NET Core in Sinhala](https://youtube.com/playlist?list=PLvvtf05eMZ2CpeAsq93DqWJHHyvCSa2Qn) - Fiqri Ismail
 
 
 ### Bootstrap
 
-* [Bootstap](https://youtube.com/playlist?list=PLXNgqM9ig24c7IdumyymD9q3e2hsz9U1m&feature=shared) - Udith Sanjaya
+* [Bootstrap](https://youtube.com/playlist?list=PLXNgqM9ig24c7IdumyymD9q3e2hsz9U1m&feature=shared) - Udith Sanjaya
 
 
 ### <a id="csharp"></a>C\#
 
 * [C# Full Course in Sinhala](https://youtube.com/playlist?list=PLvvtf05eMZ2CXD2JdZgSBgyl13ODqHOkO) - Fiqri Ismail
 
 
+### Deep Learning
```

Comment:
maybe 'artificial intelligence' would be abetter category
It's usually not helpful to add a category for a single resource. Try to find another category, or see if there are other books in our lists that would fit into your category.

Reply:
Resolved with [`01c9770`](https://github.com/EbookFoundation/free-programming-books/pull/10749/commits/01c97709642b1487d34202f5ed6db889df04bc79)